t/startfile.lua

```diff
@@ -130,19 +139,19 @@ function failover_factory(zones, local_zone)
         local res = near_zone(r)
         if res:hit() == false then
             -- example for mcp.log... Don't do this though :)
-            mcp.log("failed to find " .. r:key() .. " in zone: " .. local_zone)
+            -- mcp.log("failed to find " .. r:key() .. " in zone: " .. local_zone)
             --for _, zone in pairs(far_zones) do
             --    res = zone(r)
             local restable = mcp.await(r, far_zones, 1)
             for _, res in pairs(restable) do
                 if res:hit() then
                     --break
-                    return res
+                    return res, "failover_backup_hit"
                 end
             end
-            return restable[1]
+            return restable[1], "failover_backup_miss"
         end
-        return res -- send result back to client
+        return res, "failover_hit" -- send result back to client
```

Comment:
Fyi for the startfile.lua.

CMD_GET actually relies on the `failover_factory` since it's not mapped to the `walkall_factory`. Hence the log_req() ends up printing for every sampled GET call.

Reply:
Not sure I'm following the issue here? If you want to sample you need to use `log_reqsample`, else my example there will print every GET call. Or am I missunderstanding?

Reply:
Ah, I meant every GET call. I mistakenly thought log_req() was sampling.

In any case, we only would want to print a reqlog for GETs that failover. We aren't too interested in getting all this extra info for GETs from the nearzone.

I'll just make the necessary changes on our end.

Reply:
This change adds two API calls, `log_req` which logs unconditionally and `log_reqsample`, which has several options for how to conditionally log (check the e-mail thread for an example for reqsample). Then I added an example on how to log via wrapping routes. They're just API calls so you can use either of them however you see fit.